### PR TITLE
Drifting Turn  🏎 

### PR DESCRIPTION
**Pull request by ooookai**
This PR makes agents turn/rotate smoothly.
I am just playing the code on my computer. I hope this helps.
However, I am not sure the variables I declared are named properly. Please let me know if there is a better way.

`afterNode` : the node on the path after toNode.

I will clean the code later.

### CODE REVIEW

File: ABMobility/Agent.pde [hunk 1]
```diff
@@ -189,7 +191,11 @@ public class Agent {
     if ( newPath != null ) {
       path = newPath;
       pathIndex = path.size() - 2; // what happens if there are only two nodes?
+      if (pathIndex > 1){
```
Comment (aberke) [inline ABMobility/Agent.pde]: Is this if statement redundant to line 198? I think you can/should remove it

File: ABMobility/Agent.pde [hunk 2]
```diff
@@ -231,7 +237,8 @@ public class Agent {
         if (img != null) {
           p.pushMatrix();
           p.translate(pos.x, pos.y);
-          p.rotate(dir.heading() + PI * 0.5);
+          // p.rotate(dir.heading() + PI * 0.5);
```
Comment (aberke) [inline ABMobility/Agent.pde]: Please don’t leave in commented out code

**Review by aberke — CHANGES_REQUESTED** [review]

**Comment by aberke** [comment]
This seems like a nice idea! Thanks for doing this

**Comment by yasushisakai** [comment]
Looks good, I’m doing a bit of testing of this branch on my side too. As soon as I think it’s good I’ll merge. @ooookai thanks!

**Pull request merged**